agent/result_parser.py

```diff
@@ -61,6 +62,54 @@ def _build_technical_detail(
     return technical_detail
 
 
+def _compute_dna(
+    vulnerability_title: str,
+    vuln_location: vuln_mixin.VulnerabilityLocation | None,
+    param: str,
+) -> str | None:
+    """Compute a deterministic, debuggable DNA representation for a vulnerability.
+
+    Args:
+        vulnerability_title: The title of the vulnerability.
+        vuln_location: The location of the vulnerability.
+        param: params of vulnerability
+
+    Returns:
+        A deterministic JSON representation of the vulnerability DNA.
+    """
+    dna_data: dict[str, Any] = {"title": vulnerability_title, "param": param}
+
+    if vuln_location is not None:
+        location_dict: dict[str, Any] = vuln_location.to_dict()
+        sorted_location_dict = _sort_dict(location_dict)
+        dna_data["location"] = sorted_location_dict
+    else:
+        return None
+
+    return json.dumps(dna_data, sort_keys=True)
+
+
+def _sort_dict(d: dict[str, Any] | list[Any]) -> dict[str, Any] | list[Any]:
+    """Recursively sort lists within dictionary.
+
+    Args:
+        d: The dictionary or list to sort.
+
+    Returns:
+        A sorted dictionary or list.
+    """
+    if isinstance(d, dict):
+        return {k: _sort_dict(v) for k, v in sorted(d.items())}
```

Comment:
```suggestion
    if isinstance(d, dict):
        return {k: _sort_dict(v) for k, v in d.items()}
```
The order of iteration does not matter because `dict` type is unordered. So calling sorted here has no effect (except the wasted computation).

Reply:
I agree